### PR TITLE
upgrade: viem and wagmi to v2

### PR DESCRIPTION
**Pull request by VladimirMikulic**
Upgraded viem and wagmi to v2. (wagmi v2 40% smaller)

### CODE REVIEW

**Comment by dasanra** [comment]
Thank you for your contribution, we will review as soon as possible.

**Comment by VladimirMikulic** [comment]
Appreciate it @dasanra!

File: examples/wagmi/src/index.tsx [hunk 1]
```diff
@@ -2,17 +2,16 @@ import * as React from 'react';
...
-import { WagmiConfig, createConfig, configureChains, Connector } from 'wagmi';
-import { publicProvider } from 'wagmi/providers/public';
-import { SafeConnector } from 'wagmi/connectors/safe';
-import { InjectedConnector } from 'wagmi/connectors/injected';
-import { MetaMaskConnector } from 'wagmi/connectors/metaMask';
-import { WalletConnectConnector } from 'wagmi/connectors/walletConnect';
-
+import { WagmiProvider, createConfig, CreateConnectorFn, http } from 'wagmi';
 import { Buffer } from 'buffer';
+import { metaMask } from 'wagmi/connectors';
```
Comment (dasanra) [inline examples/wagmi/src/index.tsx:7]: I was wondering about this. They say they removed MetaMaskConnector but they still have this connector available.

https://wagmi.sh/react/guides/migrate-from-v1-to-v2#removed-metamaskconnector

**Review by dasanra — APPROVED** [review]
Thank you!

**Pull request merged**